src/portable/portable.c

```diff
@@ -572,7 +574,26 @@ static int extract_image_and_extensions(
 
         assert(name_or_path);
 
-        r = image_find_harder(IMAGE_PORTABLE, name_or_path, NULL, &image);
+        /* If we get a path, then check if it can be resolved with vpick. We need this as we might just
+         * get a simple image name, which would make vpick error out. */
+        if (path_is_absolute(name_or_path)) {
```

Comment:
i'd prefer if we'd do a more careful check:

```c
if (image_name_is_valid(…)) {
        …
else if (path_is_valid(…)) {
        …
} else
        return -EINVAL;
```

Reply:
image_find_harder already checks that